### ISSUE TITLE
Feedback Thread 🎉 

### ISSUE THREAD

**Issue by WillPapper** [wpapper/AdventureGold#1]
Hi all, this thread can be used for feedback on the Adventure Gold contract!

If you've audited it, post here to let us know! If you have general feedback around features, leave a comment on this thread as well. If your feedback is substantial, you should open a new issue or pull request.

**Comment by HickupHH3** [comment]
Hi, just a couple of comments:
1. It should be `value * 10 ** decimals()` instead of `value ** decimals()`. Numbers will become too big and overflow otherwise. Affected lines are 25, 159 and 183.
2. I don't think it's necessary, but if you want to be really safe, add ReentrancyGuard to claim methods
3. `_claim()` can take in the address of the lootId owner since it's already checked / fetched in the parent claim methods. Saves gas too, by avoiding the need to fetch it again.
4. Perhaps DAO should only be able to increment seasons. I'm not sure if there's a use case for going back to previous seasons... 
5. Claim methods (except the internal _claim method) and a couple of dao setter methods can be made `external` to save gas.

Looking forward to the launch! =)

**Comment by WillPapper** [comment]
Thank you @HickupHH3! Great catches. Really appreciate your review on this.

1. Great catch! I rarely use display values in my own smart contracts so I missed this one. Fixed!
2. I'll leave this one out for gas purposes since we already use the Checks-Effects-Interactions pattern and the only external calls are to a known/trusted smart contract. Thank you for suggesting it, if this contract grows in complexity I'll add it in.
3. Good call! I had originally kept the extra lookup in for a bit of extra reassurance (in a worst case scenario of an unknown vulnerability, someone could only mint tokens to the holder's address). Looking it over, this refactor is a low risk change so I went ahead and made it. We could have many claims so this gas optimization is worth it. Fixed!
4. I went back and forth on this one. I also don't see a use case yet for going back to previous seasons, but I do want the DAO to have the flexibility to decide. It's possible that they may want to re-open a previous claim period that passed, for example.
5. Great catch! Fixed. All methods are now external besides the internal _claim() method, `daoSetSeason()`, and `daoSetAdventureGoldPerTokenId()`.
